source/helpers/selector-observer.tsx

```diff
@@ -40,12 +41,17 @@ export default function observe<
 	registerAnimation();
 
 	const rule = document.createElement('style');
+	if (isDevelopmentVersion()) {
+		// For debuggability
+		rule.setAttribute('s', selector);
+	}
+
 	rule.textContent = css`
 		:where(${String(selector)}):not(.${seenMark}) {
 			animation: 1ms ${animation};
 		}
 	`;
-	document.head.append(rule);
+	document.body.prepend(rule);
```

Comment:
Turbo still messes with our style tags, so we just can't added to `head` or anywhere near.